crates/typos-cli/src/config.rs

```diff
@@ -155,93 +157,49 @@ pub struct TypeEngineConfig {
 
 impl TypeEngineConfig {
     pub fn from_defaults() -> Self {
-        let patterns = [
-            (
-                KString::from("lock"),
-                GlobEngineConfig {
-                    extend_glob: Vec::new(),
-                    engine: EngineConfig {
-                        check_file: Some(false),
-                        ..Default::default()
-                    },
-                },
-            ),
-            (
-                KString::from("vim"),
-                GlobEngineConfig {
-                    extend_glob: Vec::new(),
-                    engine: EngineConfig {
-                        dict: Some(DictConfig {
-                            extend_identifiers: maplit::hashmap! {
-                                "windo".into() => "windo".into(),
-                            },
+        let patterns = file_type_specifics::NO_CHECK_TYPES
+            .iter()
+            .map(|no_check_type| {
```

Comment:
This is doing too much work with iterator combiantors which I find tends to lead to less readable, less debugable, and harder to refactor code.

Reply:
Oh yeah, I should have remembered your distaste for iterator chains. I changed it to just mutate the HashMap.

Reply:
Iterator chains are fine, if they are simple.